### ISSUE TITLE
Touchmove events not being sent to HTML ad content

### ISSUE THREAD

**Issue by aduros** [mopub/mopub-android-sdk#428]
The HTML touchmove event isn't sent to the WebView contents. We'd like to use touchmove in order to build rich playable ads that use swipe and other touch gestures besides simple tapping.

For what it's worth, this works well on the iOS SDK.

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### Steps to reproduce the behavior:

1. Add this script to an HTML ad:

```js
document.body.addEventListener("touchstart", function (event) {
    console.log("DOM event: touchstart");
    event.preventDefault();
}, {passive: false});
document.body.addEventListener("touchmove", function (event) {
    console.log("DOM event: touchmove");
    event.preventDefault();
}, {passive: false});
document.body.addEventListener("touchend", function (event) {
    console.log("DOM event: touchend");
});
document.body.style.touchAction = "none";
```

2. Press and hold the screen while swiping.

#### Expected behavior:

Touchstart followed by touchmove events, then a touchend.

#### Observed behavior:

Only touchstart and touchend are logged.

#### Possible fix:

[BaseHtmlWebView:104](
https://github.com/mopub/mopub-android-sdk/blob/master/mopub-sdk/mopub-sdk-base/src/main/java/com/mopub/mobileads/BaseHtmlWebView.java#L104) consumes ACTION_MOVE events and prevents them from reaching the WebView. I believe this callback should simply be changed to always return false to allow the event to pass through.

Thanks!

**Comment by gitcatrat** [comment]
This event is extremely important for us as well. SDK does trigger `mousemove` but only once and with delay.

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

Thanks for letting us know. We will check the details of the disparity between iOS and Android and get back to you on this.

MoPub Team

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

May we ask if this is it truly HTML WebView only? Or is this on **MRAID** as well?

Please let us know.

Thanks,
Mert

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

We'd like to kindly check in with you on our [latest questions](https://github.com/mopub/mopub-android-sdk/issues/428#issuecomment-700186250). Please let us know.

Thanks,
Mert

**Comment by gitcatrat** [comment]
@hmertkahraman Sorry about the delay.

> May we ask if this is it truly HTML WebView only? Or is this on MRAID as well?

100% HTML/JS, even though we use MRAID API for some actions (e.g open url, etc) if it's available. 

> Also is this for swiping to dismiss the ad or swiping for a different use case?

We use most JS events for ad interactions, including `mousedown/touchstart`, `mousemove/touchmove`, `mouseup/touchend`, `click`, etc.

**Comment by hmertkahraman** [comment]
Thanks @gitcatrat , we'll get back to you on this.

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

Feedback and feature request is fully passed to our team. We will update here once and if there are any updates and if it'll be shipped to the SDK.

Thanks,
Mert

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

As an update: We have internal work scheduled for this, but that there is no ETA at this time on completion. We will let you know when we have a further update on this.

Thanks,
Mert

**Comment by hmertkahraman** [comment]
Hi @gitcatrat ,

Can you confirm if this ask towards HTML ads is related with **direct serve** ads only?

Thanks,
Mert

**Comment by gitcatrat** [comment]
Hey @hmertkahraman, yes you're correct.

**Comment by hmertkahraman** [comment]
H @gitcatrat ,

Checking in again to let you know that this is under our radar, but mentioned earlier there is no ETA at this time on completion.

Thanks,
Mert

**Comment by gitcatrat** [comment]
Hi @hmertkahraman,

Still in low priority TODO list? Stumbled on this issue again become somebody commented out our hack fix. We got around this by checking pointer position in tight interval from touch start until the end.

Thanks!

**Comment by chauduyphanvu** [comment]
@gitcatrat, no updates as of yet. Will update this thread - stay tuned!

**Comment by gitcatrat** [comment]
Comment for bot. 💯

**Comment by sebastiankoepke** [comment]
Hi there!
What is the current status of this issue? Is there a fix for it by now?
We also create richmedia ads with a lot of user interactions, so the touchmove events are very important for us. In the MoPub sample app, the problem still occurs with HTML ads. We have only encountered it on android.
